src/Gir/Generation/IGeneratableExtensions.cs

```diff
@@ -34,35 +35,73 @@ public static void GenerateMembers (this IGeneratable gen, IndentWriter writer,
 					doc.GenerateDocumentation(writer);
 				member.Generate(gen, writer);
 
-				if (i != array.Length - 1 && member.NewlineAfterGeneration (writer.Options))
+				if (i != array.Length - 1 && member.NewlineAfterGeneration(writer.Options))
 					writer.WriteLine();
 			}
 		}
 
-		static IEnumerable<IMemberGeneratable> GetMemberGeneratables (this IGeneratable gen)
+		static IEnumerable<IMemberGeneratable> GetMemberGeneratables(this IGeneratable gen)
 		{
 			return Utils.GetAllCollectionMembers<IMemberGeneratable>(gen);
 		}
 
-		static string GetReturnCSharpType (this ICallable callable, IndentWriter writer)
+		static string GetReturnCSharpType(this ICallable callable, IndentWriter writer)
 		{
 			// This can also be array
 			return callable.ReturnValue?.Type?.GetSymbol(writer.Options).CSharpType ?? "void";
 		}
 
-		public static void GenerateCallableDefinition (this ICallable callable, IndentWriter writer)
+		public static void GenerateCallableDefinition(this ICallable callable, IndentWriter writer)
 		{
 			callable.ReturnValue.GenerateDocumentation(writer);
 
 			writer.WriteIndent();
 			if (!string.IsNullOrEmpty(callable.Modifiers))
 				writer.Write(callable.Modifiers + " ");
 
-			var returnType = callable.GetReturnCSharpType (writer);
+			var returnType = callable.GetReturnCSharpType(writer);
 
 			// generate ReturnValue then Parameters
 			writer.Write(string.Format("{0} {1} {2}", returnType, callable.Name, "PARAMS"));
 			writer.WriteLine();
 		}
+
+		public static void GenerateConstructor(this ICallable callable, IGeneratable parent, IndentWriter writer)
+		{
+			callable.ReturnValue.GenerateDocumentation(writer);
+
+			var modifier = "";
+			if (parent is Class) {
+				if (!(parent as Class).Abstract)
+					modifier = "public ";
+				else
+					modifier = "protected ";
+			}
+
+			var (typesAndNames, names) = BuildParameters(callable.Parameters);
+
+			// FIXME, should check to see if it is deprecated
+			writer.WriteLine($"{modifier}{parent.Name}({typesAndNames}) : base ({names})");
+			writer.WriteLine("{");
+			writer.WriteLine("}");
+		}
+
+		public static (string both, string names) BuildParameters(List<Parameter> parameters)
+		{
+			var typeAndName = new Dictionary<string, string>();
+			foreach (var parameter in parameters)
+				typeAndName.Add(parameter.Type.Name, parameter.Name);
+
+			var sb = new StringBuilder();
+			foreach (var pair in typeAndName) {
+				sb.AppendFormat("{0} {1}, ", pair.Key, pair.Value);
+			}
+			string parameterString = sb.ToString();
+			parameterString = parameterString.Substring(0, parameterString.Length - 2);
```

Comment:
This will fail with no elements. I'm also not sure why to cut off everything but the last two characters. Did you wanted to cut off the trailing ", "?

Reply:
I'll open a PR.

Reply:
I do want to cut the trailing ", " because that isn't valid c#.
ie, if I have 1 parameter `string value` it'll set parameterString to `string value, `. This is just trimming off that closing `, `.

Reply:
Ah, I misread that statement but anyway it fails when there are no parameters.

Reply:
true, but this has also been rewritten in master :)

Reply:
Yeah, sorry for not making a PR. Thought I was on a branch and pushed to master :(